### ISSUE TITLE
[Solved] Error occurred while building gcc-10.2.0

### ISSUE THREAD

**Issue by guyutongxue** [niXman/mingw-builds#525]
I tried to build gcc 10.2.0 with this script, but I ran into the same problem as #520 (something about `%llu`).

Here's my build command:
```
$ ./build --mode=gcc-10.2.0 --arch=x86_64 --exceptions=seh --threads=posix
```

I tried changing `--with-default-msvcrt` option, but problem still exists. 

I even tried just define `PRIu64` macro directly to `%I64u` unconditionally, but I got these errors:
```
libtool: compile:  /home/Guyutongxue/mingw-gcc-10.2.0/x86_64-1020-posix-seh-rt_v7/build/gcc-10.2.0/./gcc/xgcc -B/home/Guyutongxue/mingw-gcc-10.2.0/x86_64-1020-posix-seh-rt_v7/build/gcc-10.2.0/./gcc/ -L/mingw64/x86_64-w64-mingw32/lib -L/mingw64/mingw/lib -isystem /mingw64/x86_64-w64-mingw32/include -isystem /mingw64/mingw/include -B/mingw64/x86_64-w64-mingw32/bin/ -B/mingw64/x86_64-w64-mingw32/lib/ -isystem /mingw64/x86_64-w64-mingw32/include -isystem /mingw64/x86_64-w64-mingw32/sys-include -DHAVE_CONFIG_H -I. -I../../../../../src/gcc-10.2.0/libgomp -I../../../../../src/gcc-10.2.0/libgomp/config/mingw32 -I../../../../../src/gcc-10.2.0/libgomp/config/posix -I../../../../../src/gcc-10.2.0/libgomp -I../../../../../src/gcc-10.2.0/libgomp/../include -Wall -pthread -Werror -g -O2 -pipe -fno-ident -I/home/Guyutongxue/mingw-gcc-10.2.0/x86_64-1020-posix-seh-rt_v7/mingw64/opt/include -I/home/Guyutongxue/mingw-gcc-10.2.0/prerequisites/x86_64-zlib-static/include -I/home/Guyutongxue/mingw-gcc-10.2.0/prerequisites/x86_64-w64-mingw32-static/include -MT env.lo -MD -MP -MF .deps/env.Tpo -c ../../../../../src/gcc-10.2.0/libgomp/env.c  -DDLL_EXPORT -DPIC -o .libs/env.o
../../../../../src/gcc-10.2.0/libgomp/env.c: In function 'handle_omp_display_env':
../../../../../src/gcc-10.2.0/libgomp/env.c:1285:24: error: format '%u' expects argument of type 'unsigned int', but argument 3 has type 'long long unsigned int' [-Werror=format=]
 1285 |       fprintf (stderr, "  GOMP_SPINCOUNT = '%"PRIu64"'\n",
      |                        ^~~~~~~~~~~~~~~~~~~~~~~
 1286 |         (uint64_t) gomp_spin_count_var);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         |
      |         long long unsigned int
In file included from ../../../../../src/gcc-10.2.0/libgomp/env.c:40:
D:/Program/msys/home/Guyutongxue/mingw-gcc-10.2.0/x86_64-1020-posix-seh-rt_v7/mingw64/mingw/include/inttypes.h:43:20: note: format string is defined here
   43 | #define PRIu64 "I64u"
```

I have no idea how to solve this...

**Comment by niXman** [comment]
@Prince213 is there any progress related to this issue?

**Comment by theodelrieu** [comment]
I have the same issue when building with gcc 9.3.0 64bit.

Command:

>./build --mode=gcc-9.3.0 --arch=x86_64 --buildroot=buildroot --threads=posix --exceptions=seh --enable-languages=c,c++

Output:

```
/bin/sh ./libtool  --tag=CC   --mode=compile /c/.conan/493c7b/1/mingw-builds/buildroot/x86_64-930-posix-seh-rt_v7/build/gcc-9.3.0/./gcc/xgcc -B/c/.conan/493c7b/1/mingw-builds/buildroot/x86_64-930-posix-seh-rt_v7/build/gcc-9.3.0/./gcc/ -L/mingw64/x86_64-w64-mingw32/lib -L/mingw64/mingw/lib -isystem /mingw64/x86_64-w64-mingw32/include -isystem /mingw64/mingw/include -B/mingw64/x86_64-w64-mingw32/bin/ -B/mingw64/x86_64-w64-mingw32/lib/ -isystem /mingw64/x86_64-w64-mingw32/include -isystem /mingw64/x86_64-w64-mingw32/sys-include    -DHAVE_CONFIG_H -I. -I../../../../../src/gcc-9.3.0/libgomp  -I../../../../../src/gcc-9.3.0/libgomp/config/mingw32 -I../../../../../src/gcc-9.3.0/libgomp/config/posix -I../../../../../src/gcc-9.3.0/libgomp -I../../../../../src/gcc-9.3.0/libgomp/../include  -Wall -Werror -Wc,-pthread  -g -O2 -pipe -fno-ident -I/c/.conan/493c7b/1/mingw-builds/buildroot/x86_64-930-posix-seh-rt_v7/mingw64/opt/include -I/c/.conan/493c7b/1/mingw-builds/buildroot/prerequisites/x86_64-zlib-static/include -I/c/.conan/493c7b/1/mingw-builds/buildroot/prerequisites/x86_64-w64-mingw32-static/include -MT target.lo -MD -MP -MF .deps/target.Tpo -c -o target.lo ../../../../../src/gcc-9.3.0/libgomp/target.c
libtool: compile:  /c/.conan/493c7b/1/mingw-builds/buildroot/x86_64-930-posix-seh-rt_v7/build/gcc-9.3.0/./gcc/xgcc -B/c/.conan/493c7b/1/mingw-builds/buildroot/x86_64-930-posix-seh-rt_v7/build/gcc-9.3.0/./gcc/ -L/mingw64/x86_64-w64-mingw32/lib -L/mingw64/mingw/lib -isystem /mingw64/x86_64-w64-mingw32/include -isystem /mingw64/mingw/include -B/mingw64/x86_64-w64-mingw32/bin/ -B/mingw64/x86_64-w64-mingw32/lib/ -isystem /mingw64/x86_64-w64-mingw32/include -isystem /mingw64/x86_64-w64-mingw32/sys-include -DHAVE_CONFIG_H -I. -I../../../../../src/gcc-9.3.0/libgomp -I../../../../../src/gcc-9.3.0/libgomp/config/mingw32 -I../../../../../src/gcc-9.3.0/libgomp/config/posix -I../../../../../src/gcc-9.3.0/libgomp -I../../../../../src/gcc-9.3.0/libgomp/../include -Wall -pthread -Werror -g -O2 -pipe -fno-ident -I/c/.conan/493c7b/1/mingw-builds/buildroot/x86_64-930-posix-seh-rt_v7/mingw64/opt/include -I/c/.conan/493c7b/1/mingw-builds/buildroot/prerequisites/x86_64-zlib-static/include -I/c/.conan/493c7b/1/mingw-builds/buildroot/prerequisites/x86_64-w64-mingw32-static/include -MT target.lo -MD -MP -MF .deps/target.Tpo -c ../../../../../src/gcc-9.3.0/libgomp/target.c  -DDLL_EXPORT -DPIC -o .libs/target.o
../../../../../src/gcc-9.3.0/libgomp/target.c: In function 'gomp_map_vars':
../../../../../src/gcc-9.3.0/libgomp/target.c:936:21: error: unknown conversion type character 'l' in format [-Werror=format=]
  936 |         gomp_fatal ("present clause: !acc_is_present (%p, "
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../../../../../src/gcc-9.3.0/libgomp/target.c:37:
C:/.conan/493c7b/1/mingw-builds/buildroot/x86_64-930-posix-seh-rt_v7/mingw64/mingw/include/inttypes.h:36:18: note: format string is defined here
   36 | #define PRIu64 "llu"
      |                  ^
../../../../../src/gcc-9.3.0/libgomp/target.c:936:21: error: unknown conversion type character 'l' in format [-Werror=format=]
  936 |         gomp_fatal ("present clause: !acc_is_present (%p, "
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../../../../../src/gcc-9.3.0/libgomp/target.c:37:
C:/.conan/493c7b/1/mingw-builds/buildroot/x86_64-930-posix-seh-rt_v7/mingw64/mingw/include/inttypes.h:37:18: note: format string is defined here
   37 | #define PRIx64 "llx"
      |                  ^
../../../../../src/gcc-9.3.0/libgomp/target.c:936:21: error: too many arguments for format [-Werror=format-extra-args]
  936 |         gomp_fatal ("present clause: !acc_is_present (%p, "
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1.exe: all warnings being treated as errors
```

**Comment by theodelrieu** [comment]
I guess I'll patch the script to remove Werror...

**Comment by theodelrieu** [comment]
Using `--bootstrap` fixes the issue, but it builds a whole lot a stuff

**Comment by niXman** [comment]
I think we should make `--bootstrap` as default option (which will be disabled when `--no-bootstrap` is specified) because there are too many problems with this...

can you please provide such a fix?

**Comment by theodelrieu** [comment]
@niXman Well, my bash skills are quite limited but I'll try.

EDIT: Nevermind, I have no idea how to do this arcanic stuff. I'll have to pass on that one, sorry!

**Comment by niXman** [comment]
got it, will do later: https://github.com/niXman/mingw-builds/issues/527

**Comment by guyutongxue** [comment]
Sorry, I still can't build successfully with `--bootstrap` option on. :confused:

Command:
```
./build --mode=gcc-10.2.0 --arch=x86_64 --exceptions=seh --threads=posix --jobs=6 --bootstrap --enable-languages=c,c++
```
Error:
```
In file included from ../../../../../src/gcc-10.2.0/libgomp/oacc-parallel.c:30:
../../../../../src/gcc-10.2.0/libgomp/oacc-parallel.c: In function 'GOACC_parallel_keyed':
../../../../../src/gcc-10.2.0/libgomp/oacc-parallel.c:122:18: error: unknown conversion type character 'l' in format [-Werror=format=]
  122 |   gomp_debug (0, "%s: mapnum=%"PRIu64", hostaddrs=%p, size=%p, kinds=%p\n",
      |                  ^~~~~~~~~~~~~~
../../../../../src/gcc-10.2.0/libgomp/libgomp.h:185:29: note: in definition of macro 'gomp_debug'
  185 |       (gomp_debug) ((KIND), __VA_ARGS__); \
      |                             ^~~~~~~~~~~
```
([Full log](https://paste.ubuntu.com/p/ZJ5mmNpgYy/))

@theodelrieu Did you do anything other than add `--bootstrap` option when you successfully built it?

**Comment by theodelrieu** [comment]
@Guyutongxue I'm building GCC 9.3.0, maybe it's specific to 10?

**Comment by guyutongxue** [comment]
> @Guyutongxue I'm building GCC 9.3.0, maybe it's specific to 10?

I've tried GCC 9.3.0 too, but those errors still exist:
```
../../../../../src/gcc-9.3.0/libgomp/target.c: In function 'gomp_map_vars':
../../../../../src/gcc-9.3.0/libgomp/target.c:936:21: error: unknown conversion type character 'l' in format [-Werror=format=]
  936 |         gomp_fatal ("present clause: !acc_is_present (%p, "
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../../../../../src/gcc-9.3.0/libgomp/target.c:37:
D:/Program/msys64/home/Guyutongxue/mingw-gcc-9.3.0/x86_64-930-posix-seh-rt_v7/mingw64/mingw/include/inttypes.h:36:18: note: format string is defined here
   36 | #define PRIu64 "llu"
      |                  ^
../../../../../src/gcc-9.3.0/libgomp/target.c:936:21: error: unknown conversion type character 'l' in format [-Werror=format=]
  936 |         gomp_fatal ("present clause: !acc_is_present (%p, "
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../../../../../src/gcc-9.3.0/libgomp/target.c:37:
D:/Program/msys64/home/Guyutongxue/mingw-gcc-9.3.0/x86_64-930-posix-seh-rt_v7/mingw64/mingw/include/inttypes.h:37:18: note: format string is defined here
   37 | #define PRIx64 "llx"
      |                  ^
../../../../../src/gcc-9.3.0/libgomp/target.c:936:21: error: too many arguments for format [-Werror=format-extra-args]
  936 |         gomp_fatal ("present clause: !acc_is_present (%p, "
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1.exe: all warnings being treated as errors
```

**Comment by theodelrieu** [comment]
Weird, did you erase everything else from previous builds?

**Comment by guyutongxue** [comment]
Not. I'll try again later when I've cleaned up, thanks for the reminder.

**Comment by guyutongxue** [comment]
Unfortunately, there are still same errors. 
**EDIT:**
The final workaround is to add `-Wno-format` option when compiling C.
```
./build --mode=gcc-10.2.0 --arch=x86_64 --exceptions=seh --threads=posix --jobs=6 --cflags='-fno-builtin -Wno-format'
```
It's stupid, but it works.

**Comment by niXman** [comment]
>The final workaround is to add -Wno-format option when compiling C.

hmm...

**Comment by revelator** [comment]
The problem comes from libgomp hardcoding the printf format, theres a patch that gets around that on the mingw-packages site.

**Comment by Prince213** [comment]
Fixed in #532 . No workaround is needed now.
